Compatibility issue with Kotlin 1.6

**Describe the bug**
Duration API has been made stable in Kotlin 1.6.0, and some methods have been deleted. As result, the following error appears:

```
java.lang.NoSuchMethodError: No static method toDuration(DLjava/util/concurrent/TimeUnit;)J in class Lkotlin/time/DurationKt; or its super classes (declaration of 'kotlin.time.DurationKt' appears in /data/app/com.enki.Enki750g-dL2OkPW0zzwa1CQrnIAsnQ==/base.apk!classes22.dex)
        at com.dropbox.android.external.store4.MemoryPolicy.<clinit>(MemoryPolicy.kt:116)
        at com.dropbox.android.external.store4.StoreDefaults.<clinit>(StoreDefaults.kt:23)
        at com.dropbox.android.external.store4.RealStoreBuilder.<init>(StoreBuilder.kt:91)
        at com.dropbox.android.external.store4.StoreBuilder$Companion.from(StoreBuilder.kt:76)
        at com.webedia.food.config.ConfigManager.<init>(ConfigManager.kt:51)
```

**To Reproduce**

Create a store with the default options.

**Expected behavior**
Should not crash

**Infos**
 - Store Version: 4.0.2-KT15

Hi folks, store has modules that depend on Moshi, we are unable to upgrade Store to 1.6 until Moshi metadata issues are resolved 
Caused by: java.lang.IllegalStateException: Could not parse metadata! This should only happen if you're using Kotlin <1.1.
        at com.squareup.moshi.kotlinpoet.metadata.KotlinPoetMetadata.readKotlinClassMetadata(KotlinPoetMetadata.kt:89)

Will keep this open for now

--- a/store/src/main/java/com/dropbox/android/external/store4/MemoryPolicy.kt
+++ b/store/src/main/java/com/dropbox/android/external/store4/MemoryPolicy.kt
@@ -1,9 +1,7 @@
 package com.dropbox.android.external.store4
 
-import java.util.concurrent.TimeUnit
 import kotlin.time.Duration
 import kotlin.time.ExperimentalTime
-import kotlin.time.toDuration
 
 fun interface Weigher<in K : Any, in V : Any> {
     /**
@@ -109,11 +107,7 @@ class MemoryPolicy<in Key : Any, in Value : Any> internal constructor(
     }
 
     companion object {
-        // Ideally this would be set to Duration.INFINITE, but this breaks consumers compiling with
-        // Kotlin 1.4-M3 (not sure why).
-        // TODO: revert back to using Duration.INFINITE once Store is compiled with Kotlin 1.4
-        val DEFAULT_DURATION_POLICY: Duration =
-            Double.POSITIVE_INFINITY.toDuration(TimeUnit.SECONDS)
+        val DEFAULT_DURATION_POLICY: Duration = Duration.INFINITE
         const val DEFAULT_SIZE_POLICY: Long = -1
 
         fun <Key : Any, Value : Any> builder(): MemoryPolicyBuilder<Key, Value> =

